.github/workflows/rust.yml

```diff
@@ -139,3 +139,24 @@ jobs:
           $env:VCPKG_ROOT = $env:VCPKG_INSTALLATION_ROOT
           cargo build
           cargo test
+  windows-arm64:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: rust
+      - name: Install Protoc v21.12
+        working-directory: C:\
+        run: |
+          New-Item -Path 'C:\protoc' -ItemType Directory
+          Set-Location C:\protoc
+          Invoke-WebRequest https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-win64.zip -OutFile C:\protoc\protoc.zip
+          7z x protoc.zip
+          Add-Content $env:GITHUB_PATH "C:\protoc\bin"
+        shell: powershell
+      - name: Run tests
+        run: |
+          $env:VCPKG_ROOT = $env:VCPKG_INSTALLATION_ROOT
+          cargo build --target aarch64-pc-windows-msvc
+          cargo test --target aarch64-pc-windows-msvc
```

Comment:
I don't think this will work since the `windows-latest` runner is x86_64.

Reply:
It doesn't look like there is a OSS/free option for arm Windows runner. That does make me worry about maintaining this https://github.com/actions/runner-images/issues/10820

Reply:
I know :( I was quite confused with GH Runner docs... I think `lancedb` org needs to create a windows arm runner for it to work.

Reply:
Hey @wjones127 What should we do? What are your thoughts?

Reply:
I've setup a runner we can use to test arm64 windows.